### PR TITLE
Transaction confirmation modal

### PR DESCRIPTION
**Pull request by wbobeirne**
## What This Is

This addresses #85, and adds a confirmation modal to the send tab. There's still a few missing pieces, and this PR comes with a lot of `// TODO: ...`, but it's at least a start. You can also see a video of it in action [over here](https://gfycat.com/RemoteSelfassuredDove).

## What I Did

* Created and styled a modal for confirming transactions
* Made the modal use the actual transaction object, _not_ what's in the form fields
* Reworked some of the copy and added some additional data (Translation issues?)
* Added a 5 second timer that hopefully gets people to stop and read
* Refactored some unit conversion code, wrote unit tests for it
* Fixed up / added some flow typing

## What I Didn't Do

* Implement _all_ of the bullets in #85. Much of that doesn't exist in the current modal, so I wasn't sure how much we were adding.
* Handle any non-Ethereum transactions (Showing altcoin instead of ETH.) I may need some assistance on how this is handled elsewhere right now.
* Nothing currently generates transactions (See #100) so that's not a part of this. There's no way to actually get this modal to open naturally
* AFAIK there's no infrastructure in the codebase right now for broadcasting a transaction, so the confirm button doesn't work

## Comparison

Old:
old

New: 
new

### CODE REVIEW

File: common/components/ui/Modal.jsx [hunk 1]
```diff
@@ -18,6 +17,8 @@ type Props = {
       | 'warning'
       | 'danger'
       | 'link',
+    disabled?: boolean,
+    // $FlowFixMe - Why the fuck doesn't this like onClick?
```
Comment (wbobeirne) [inline common/components/ui/Modal.jsx]: If anyone could help me out with this, it doesn't like the onClicks I provided in ConfirmationModal.jsx and the error is really opaque, just saying my buttons don't match the button object.

File: common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import translate from 'translations';
+import Big from 'bignumber.js';
+import BaseWallet from 'libs/wallet/base';
+import { toUnit } from 'libs/units';
+import type { NodeConfig } from 'config/data';
+
+import Modal from 'components/ui/Modal';
+import Identicon from 'components/ui/Identicon';
+
+// TODO: Replace me with a shared type defined somewhere
```
Comment (wbobeirne) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx]: Likely as a part of #100.

File: common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+export default class ConfirmationModal extends React.Component {
+  props: Props;
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+    const { value, gasPrice, nonce } = props.rawTransaction;
+
+    // TODO: Fix with wallet address promise
```
Comment (wbobeirne) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx]: As a part of #105.

File: common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            </li>
+            <li className="ConfModal-details-detail">
+              You are transmitting on the <strong>{node.network}</strong>{' '}
+              network provided by <strong>{node.service}</strong>
+            </li>
+            <li className="ConfModal-details-detail">
+              {rawTransaction.data
+                ? <span>
+                    You are sending the following data:{' '}
+                    <code>{rawTransaction.data}</code>
```
Comment (wbobeirne) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx]: How big does data get? Should I find a better display for this?

File: common/containers/Tabs/SendTransaction/components/ConfirmationModal.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    &-amount {
+      flex-grow: 1;
+      text-align: center;
+      line-height: $summary-height / 2;
+
+      &-arrow {
+        display: inline-block;
+        height: $summary-height / 3;
+        width: 100%;
+        background-image: url('../../../../assets/images/icon-dot-arrow.svg');
```
Comment (wbobeirne) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.scss]: Remind me to fix this to not use relative paths.

**Comment by tayvano** [comment]
The only thing I would request is adding the from address. I figured people would recognize their own sending address icon. I was wrong. Here is what the current site looks like now—I'm sure there is a better way. :P 
screen shot 2017-08-07 at 6 05 20 pm

**Comment by wbobeirne** [comment]
Did this just get added? Wasn't in my screenshot of the before, and I took it like minutes before submitting the PR!

But yeah, I'd be happy to add that in.

File: common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+          <ul className="ConfModal-details">
+            <li className="ConfModal-details-detail">
+              You are sending to <code>{rawTransaction.to}</code>
+            </li>
+            <li className="ConfModal-details-detail">
+              You are sending <strong>{value} ETH</strong> with a gas price of{' '}
+              <strong>{gasPrice} gwei</strong>
+            </li>
+            <li className="ConfModal-details-detail">
+              You are transmitting on the <strong>{node.network}</strong>{' '}
```
Comment (wbobeirne) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx]: Not sure where I came up with this copy, original is `interacting with the` which sounds way less nerdy. I'll change it back.

Comment (tayvano) [inline common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx]: @wbobeirne https://appear.in/myetherwallet

**Comment by wbobeirne** [comment]
Added in `from` to the bullet list. Integrated the generate transaction code from #100, so now this actually pops up. I'm going to take `WIP` off the title, I think this is good to go.

**Comment by crptm** [comment]
This should take signed tx

**Comment by wbobeirne** [comment]
I definitely agree in the long term, what would you think about merging this in as-is and making a new issue for that? It's already somewhat better than v3 which just read form inputs, and doesn't change the user experience.

Not downplaying the value of reading from signed though, I really want to make that happen. It would probably just take me a little while to figure out how to convert from that to the raw data, and I want to get in #118 because a lot of work towards handling that is done in there.

**Comment by crptm** [comment]
We should have started with isolated modal that takes rawtx string in the first place. Ethereumjs-tx can parse signed txes.

**Comment by crptm** [comment]
We can merge this as is bc send token PR is on top of this branch.

**Comment by wbobeirne** [comment]
#118 has been merged in, and the modal takes signed transactions now (Along with all the other token handling.) Are there any other outstanding issues?

**Comment by crptm** [comment]
If we only support EIP-155 chains is it possible to check that we broadcast tx on correct network? This could wait for other PR, just curious if this is possible. 

PR should be merged, no reason to make it any bigger.

**Comment by wbobeirne** [comment]
I don't know enough about the alt chains to know that, but presumably we could make that kind of confirmation optional iff they provide chainid right?

Can somebody throw me a 👍 so that I can merge this?

**Comment by tayvano** [comment]
Can you confirm that it shows the sending from address somewhere (underneath the addressicon? This has become an issue of late. Hold on EIP-155 stuff until I get everything sorted with v3 issues and we can circle back. I dont think it applies here regardless.

**Pull request merged**